File: z-clients/javascript/src/apis/adminCluster.ts
// this file is @generated

import {
    type ClusterForceSnapshotIn,
    ClusterForceSnapshotInSerializer,
} from '../models/clusterForceSnapshotIn';
import {
    type ClusterForceSnapshotOut,
    ClusterForceSnapshotOutSerializer,
} from '../models/clusterForceSnapshotOut';
import {
    type ClusterInitializeIn,
    ClusterInitializeInSerializer,
} from '../models/clusterInitializeIn';
import {
    type ClusterInitializeOut,
    ClusterInitializeOutSerializer,
} from '../models/clusterInitializeOut';
import {
    type ClusterRemoveNodeIn,
    ClusterRemoveNodeInSerializer,
} from '../models/clusterRemoveNodeIn';
import {
    type ClusterRemoveNodeOut,
    ClusterRemoveNodeOutSerializer,
} from '../models/clusterRemoveNodeOut';
import {
    type ClusterStatusOut,
    ClusterStatusOutSerializer,
} from '../models/clusterStatusOut';
import { HttpMethod, CoyoteRequest, type CoyoteRequestContext } from "../request";

export class AdminCluster {
    public constructor(private readonly requestCtx: CoyoteRequestContext) {}

    /** Get information about the current cluster */
    public status(
    ): Promise<ClusterStatusOut> {
        const request = new CoyoteRequest(HttpMethod.GET, "/api/v1.admin.cluster.status");

        
        return request.send(
            this.requestCtx,
            ClusterStatusOutSerializer._fromJsonObject,
        );
    }/**
* Initialize this node as the leader of a new cluster
* 
* This operation may only be performed against a node which has not been
* initialized and is not currently a member of a cluster.
*/
    public initialize(
        clusterInitializeIn: ClusterInitializeIn,
    ): Promise<ClusterInitializeOut> {
        const request = new CoyoteRequest(HttpMethod.POST, "/api/v1.admin.cluster.initialize");

        request.setBody(
            ClusterInitializeInSerializer._toJsonObject(clusterInitializeIn)
        );
        
        return request.send(
            this.requestCtx,
            ClusterInitializeOutSerializer._fromJsonObject,
        );
    }/**
* Remove a node from the cluster.
* 
* This operation executes immediately and the node must be wiped and reset
* before it can safely be added to the cluster.
*/
    public removeNode(
        clusterRemoveNodeIn: ClusterRemoveNodeIn,
    ): Promise<ClusterRemoveNodeOut> {
        const request = new CoyoteRequest(HttpMethod.POST, "/api/v1.admin.cluster.remove-node");

        request.setBody(
            ClusterRemoveNodeInSerializer._toJsonObject(clusterRemoveNodeIn)
        );
        
        return request.send(
            this.requestCtx,
            ClusterRemoveNodeOutSerializer._fromJsonObject,
        );
    }/** Force the cluster to take a snapshot immediately */
    public forceSnapshot(
        clusterForceSnapshotIn: ClusterForceSnapshotIn,
    ): Promise<ClusterForceSnapshotOut> {
        const request = new CoyoteRequest(HttpMethod.POST, "/api/v1.admin.cluster.force-snapshot");

        request.setBody(
            ClusterForceSnapshotInSerializer._toJsonObject(clusterForceSnapshotIn)
        );
        
        return request.send(
            this.requestCtx,
            ClusterForceSnapshotOutSerializer._fromJsonObject,
        );
    }
}

File: z-clients/python/coyote/apis/admin_cluster.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    ClusterInitializeIn,
    ClusterInitializeOut,
    ClusterRemoveNodeIn,
    ClusterRemoveNodeOut,
    ClusterStatusOut,
)


class AdminClusterAsync(ApiBase):
    async def status(
        self,
    ) -> ClusterStatusOut:
        """Get information about the current cluster"""

        return await self._request_asyncio(
            method="get",
            path="/api/v1.admin.cluster.status",
            response_type=ClusterStatusOut,
        )

    async def initialize(
        self,
        cluster_initialize_in: ClusterInitializeIn = ClusterInitializeIn(),
    ) -> ClusterInitializeOut:
        """Initialize this node as the leader of a new cluster

        This operation may only be performed against a node which has not been
        initialized and is not currently a member of a cluster."""
        body = cluster_initialize_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.admin.cluster.initialize",
            body=body,
            response_type=ClusterInitializeOut,
        )

    async def remove_node(
        self,
        cluster_remove_node_in: ClusterRemoveNodeIn,
    ) -> ClusterRemoveNodeOut:
        """Remove a node from the cluster.

        This operation executes immediately and the node must be wiped and reset
        before it can safely be added to the cluster."""
        body = cluster_remove_node_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.admin.cluster.remove-node",
            body=body,
            response_type=ClusterRemoveNodeOut,
        )


class AdminCluster(ApiBase):
    def status(
        self,
    ) -> ClusterStatusOut:
        """Get information about the current cluster"""

        return self._request_sync(
            method="get",
            path="/api/v1.admin.cluster.status",
            response_type=ClusterStatusOut,
        )

    def initialize(
        self,
        cluster_initialize_in: ClusterInitializeIn = ClusterInitializeIn(),
    ) -> ClusterInitializeOut:
        """Initialize this node as the leader of a new cluster

        This operation may only be performed against a node which has not been
        initialized and is not currently a member of a cluster."""
        body = cluster_initialize_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.admin.cluster.initialize",
            body=body,
            response_type=ClusterInitializeOut,
        )

    def remove_node(
        self,
        cluster_remove_node_in: ClusterRemoveNodeIn,
    ) -> ClusterRemoveNodeOut:
        """Remove a node from the cluster.

        This operation executes immediately and the node must be wiped and reset
        before it can safely be added to the cluster."""
        body = cluster_remove_node_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.admin.cluster.remove-node",
            body=body,
            response_type=ClusterRemoveNodeOut,
        )
